### DISCUSSION TITLE
OpenHD svpcom Audio transmission

### DISCUSSION THREAD

**Discussion by XlashDev** [svpcom/wfb-ng#137] [Questions]
I'm using OpenHD which uses this script wifibroadcast. I did ask on their telegram community but have no answers and maybe they are still busy with some other parts of the project so I will try to ask it here.

Gnd has RC  control  via joystick connected thru USB as HID
Air has video, audio , mavlink/telemetry

What I am trying to do is enabling the audio from gnd to air but the RC control freezes and vehicle puts on hold. Video , audio and telemetry still works.

This is the transmission from gnd to air part on the script

`./wfb_tx -u 5050 -p 66 -B 20 -M 0 -K /tmp/tx.key $NICS_LIST`

the air to gnd that I copied uses -u 5051 -p67

Is what I'm doing ( uplink audio  = gnd -> air) possible via wfb?  

TIA

**Comment by svpcom** [comment]
How do you transmit RC control from ground to drone? Do you embed it into mavlink telemetry or use separate data channel?
You can transmit audio and video back to the drone, but you need to take into account that radio bandwidth is shared between TX and RX. If you use MCS#1 modulation you will have max ~7mbit for both ground and drone and hd video is aboud 4mbit. So it is not possible to have two hd streams at the same time.

**Reply by XlashDev** [reply]
Thank you for swift reply. I have a hard time finding the telemetry script. But I found it I think here is a part of the c code.

> printf("\nUsage: tx_telemetry [options] <interfaces>\n\n"
           "Options:\n"
           "-c <cts>    0 = CTS protection disabled. 1 = CTS protection enabled (Atheros only)\n"
           "-p <port>   Port. Default = 1\n"
           "-r <count>  Retransmission count. 1 = send each frame once, 2 = twice, 3 = three times ... Default = 1\n"
           "-x <type>   Telemetry protocol to be used. 0 = Mavlink. 1 = generic (for all others).\n"
           "-d <rate>   Data rate to send frames with. Currently only supported with Ralink cards. Choose 6,12,18,24,36 Mbit\n"
           "-y <mode>   Transmission mode. 0 = send on all interfaces, 1 = send only on interface with best RSSI\n"
           "-z <debug>  Debug. 0 = disable debug output, 1 = enable debug output\n\n"
           "-f <force>  Force data frames. 0 = disable, 1 = use data frames on realtek cards\n\n"
           "Example:\n"
           "  cat /dev/serial0 | tx_telemetry -c 0 -p 1 -r 1 -x 0 -d 24 -y 0 wlan0\n"
           "\n");
    exit(1);

    
    and here is the rc part
    
    `/usr/local/share/cameracontrol/IPCamera/svpcom_wifibroadcast/wfb_rx_rc -c 127.0.0.1 -k 1 -n 1 -u 5565 -p 97 -K /tmp/rx.key $NICS_LIST`
    
so the rc channel is separate from the telemetry link
    
here is the project github link [https://github.com/OpenHD/Open.HD](url)
    
I only want to transmit audio back to the AIR (video not included). Even if it has low quality as long as it is working and works along with rc control. I tried lowering the bandwidth but upon testing minimum is 20 since lower value doesnt work.

**Reply by svpcom** [reply]
According to usage output it is not my code. It seems that it was rewritten by OpenHD team. I only support my original code.

**Reply by XlashDev** [reply]
> According to usage output it is not my code. It seems that it was rewritten by OpenHD team. I only support my original code.

Ok I see. Maybe that is why it is not functioning as it should be since it is modified. I understand. Thank you for your time and reply.
